internal/node/node.bzl

```diff
@@ -85,6 +85,8 @@ def _nodejs_binary_impl(ctx):
     for d in ctx.attr.data:
       if hasattr(d, "node_sources"):
         sources += d.node_sources.to_list()
+      if hasattr(d, "files"):
```

Comment:
Note: this does have to be an "if" and not an "elif" to fix `//internal/npm_package/test:test` when preserve-symlinks is turned on

Reply:
This makes me think that other code that has
```
  for d in ctx.attr.deps:
    if hasattr(d, "node_sources"):
      files = depset(transitive=[files, d.node_sources])
    elif hasattr(d, "files"):
      files = depset(transitive=[files, d.files])
```
like in `devmode_js_sources.bzl` and also in rules_typescript `ts_devserver.bzl` and `ts_webtest.bzl` in incorrect